CHANGELOG.md

```diff
@@ -1,6 +1,129 @@
 # Qurro changelog
 
-## Qurro 0.2.1 (in development)
+## Qurro 0.3.0
+### Features added
+- Now, Songbird no longer needs to be installed in order for the `qiime qurro
+  supervised-rank-plot` action (**NOTE: this action has been renamed to `qiime
+  qurro differential-plot`, see below**) to be available.
+  ([#154](https://github.com/biocore/qurro/issues/154))
+- Feature ranking values (i.e. differentials or loadings) are now shown in the
+  tooltips for individual features in the rank plot.
+  ([#186](https://github.com/biocore/qurro/issues/186))
+- Added feature rankings as searchable fields in the "Selecting Features"
+  controls. ([#97](https://github.com/biocore/qurro/issues/97))
+- Feature metadata fields and feature ranking fields are now grouped under
+  `Feature Metadata` or `Feature Ranking` headers in the searchable fields
+  dropdowns. ([#191](https://github.com/biocore/qurro/issues/191))
+- **Added numeric searching**: now you can search through numeric feature
+  metadata or feature rankings using basic comparison operators. (Non-numeric
+  input search text will result in the search not identifying any features,
+  and non-numeric feature metadata values for the specified field will just
+  be ignored in searching.)
+  ([#141](https://github.com/biocore/qurro/issues/141),
+  [#97](https://github.com/biocore/qurro/issues/97))
+- Added a `--version` parameter to the standalone (i.e. outside of QIIME 2)
+  Qurro interface. You can now check the currently installed Qurro version by
+  running `qurro --version`. (There's already a `--version` parameter for the
+  QIIME 2 Qurro plugin; this can be accessed using `qiime qurro --version`.)
+- Sample metadata fields are now ordered alphabetically (and ignoring case) in
+  the `x-axis field` and `color field` dropdowns in the sample plot controls.
+  ([#76](https://github.com/biocore/qurro/issues/76))
+- The fractions of selected features out of the total amount of features in the
+  rank plot are now displayed in the numerator and denominator "headers."
+  ([#187](https://github.com/biocore/qurro/issues/187))
+- Replaced the `Bar width` dropdown with a slider input that lets you easily
+  adjust the bar width (in addition to a checkbox that controls whether or not
+  the bar widths are fitted to the available width).
+    - The range of this slider is 1 pixel per bar to 10 pixels per bar. Note
+      that the maximum bar width (when not using the "Fit bar widths" option)
+      has been increased from 3 pixels to 10 pixels.
+    - Renamed the `Fit bar widths to display width` option to say `Fit bar
+      widths to the plot's default width?`. This is a more accurate description
+      of what this option does.
+- More output messages are now used when running Qurro. In particular, the
+  following events will now trigger output messages:
+    - Empty samples and/or features being automatically removed
```

Comment:
How are mixed types handled? For example, if an expected numeric column has missing or string values?

Reply:
Features with values that can't be read as numeric characters are ignored in numeric searching. This is tested [here](https://github.com/fedarko/qurro/blob/e774255240ef1318c04f6dabfee872ec40c409f7/qurro/tests/web_tests/tests/test_filter_features.js#L511) -- to make a long story short, what these tests are doing is just trying various numeric search options / search values against the `x` field of `rpJSON1`.